nbformat/tests/test_validator.py

```diff
@@ -101,4 +102,11 @@ def test_iter_validation_error(self):
         assert {e.ref for e in errors} == {'markdown_cell', 'heading_cell', 'bad stream'}
 
     def test_iter_validation_empty(self):
-        assert list(iter_validate({})) == list()
\ No newline at end of file
+        """Test that an empty notebook (invalid) fails validation via iter_validate"""
+        with self.assertRaises(ValidationError) as e:
+            list(iter_validate({}))
```

Comment:
I think `iter_validate()` should yield the error, not raise it. This is an odd way to use exceptions, but it's how `iter_validate()` is meant to work, so that calling code can collect and show multiple validation errors.

Reply:
OK, seems reasonable. I've updated the error handler and the test.